src/components/anchor-navigation/anchor-navigation.component.js

```diff
@@ -0,0 +1,133 @@
+import React, {
+  useState, useRef, useEffect, useCallback
+} from 'react';
+import PropTypes from 'prop-types';
+import throttle from 'lodash/throttle';
+
+import Event from '../../utils/helpers/events';
+import { StyledAnchorNavigation, StyledNavigation, StyledContent } from './anchor-navigation.style';
+import AnchorNavigationItem from './anchor-navigation-item.component';
+
+const AnchorNavigation = ({ children, stickyNavigation, styleOverride }) => {
+  const [selectedIndex, setSelectedIndex] = useState(0);
+
+  const sectionRefs = useRef(React.Children.map(stickyNavigation.props.children, child => child.props.target));
+
+  const anchorRefs = useRef(Array.from({
+    length: React.Children.count(stickyNavigation.props.children)
+  }, () => React.createRef()));
+
+  const contentRef = useRef();
+
+  const navigationRef = useRef();
+
+  const setSelectedAnchorBasedOnScroll = useCallback(throttle(() => {
```

Comment:
Ideally we'd not use `lodash` here as the aim is to hopefully remove it as a dependency in the future. You could replace it with your `throttle` own implementation like what's discussed here https://stackoverflow.com/a/55974034/10894881

Reply:
@edleeks87 It seems that our library depends on the lodash pretty extensively. Are planning to rewrite methods like `merge` or `deepClone` in the future just to get rid of the `lodash`?

Reply:
Good point, I think we need to review how we import lodash as discussed on slack, but that is out of scope for this